.gitlab-ci.yml

```
@@ -422,13 +422,17 @@ workflow:
   - <<: *if_deploy
     when: on_failure
 
-# rule to trigger job for internal image deployment if deploy is set or
-# manually if not
-.on_deploy_internal_or_manual:
+# rule to trigger job for internal image deployment if deploy is set, or if there are changes to the internal image
+# deployment config file or manually if not
+.on_deploy_internal_or_internal_image_change_or_manual:
   - !reference [.except_mergequeue]
   - <<: *if_deploy
     variables:
       RELEASE_PROD: "true"
+  - changes:
+      paths:
+        - .gitlab/internal_image_deploy/internal_image_deploy.yml
+      compare_to: $COMPARE_TO_BRANCH
```

```suggestion
      compare_to: $COMPARE_TO_BRANCH
    variables:
      RELEASE_PROD: "false"
```
I'm not sure what is the expected value in this situation but I suppose we have to set it.